query-engine/connector-test-kit-rs/query-tests-setup/src/connector_tag/js/external_process.rs

```diff
@@ -1,20 +1,66 @@
 use super::*;
 use serde::de::DeserializeOwned;
+use serde_json::Value;
+use std::fmt::Formatter;
 use std::{
     error::Error as StdError,
     fmt::Display,
     io::Write as _,
     sync::{atomic::Ordering, Arc, LazyLock},
 };
+use tokio::sync::oneshot::error::RecvError;
 use tokio::sync::{mpsc, oneshot, RwLock};
 
-type Result<T> = std::result::Result<T, Box<dyn std::error::Error + Send + Sync>>;
+#[derive(Debug, serde::Deserialize)]
+#[serde(tag = "type", content = "args")]
+enum RpcResponse<T> {
+    None(()),
+    Result(T),
+    Error(RpcError),
+}
+
+#[derive(Debug, serde::Deserialize)]
+struct RpcError {
+    message: String,
+    #[serde(default)]
+    code: Option<String>,
+    #[serde(default)]
+    stack: Option<String>,
+}
+
+#[derive(Debug)]
+pub enum Response<T> {
```

Comment:
Defining this type required a lot of code changes. Maybe it was overkill.

Should I go for `Result<Option<T>>` instead, which would require less changes?

Reply:
I would probably just go with `type Response<T> = Result<Option<T>, Box<dyn StdError + Send + Sync>>` indeed. Both can encode the exact same number of states, but the composition of `Result` and `Option` comes with a bunch of methods for free. If rather needed our own custom methods instead (which I don't see any defined), then defining a custom enum might've been worth it.

Reply:
Thanks. I agree and will use that type then.